# Анализ ошибок бейзлайна

In [1]:
import pandas as pd
import sys
import os

# Добавляем путь к проекту, чтобы можно было импортировать из utils
try:
    CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
except NameError:
    CURRENT_DIR = os.getcwd()
    
BASE_DIR = os.path.abspath(os.path.join(CURRENT_DIR, ".."))
if BASE_DIR not in sys.path:
    sys.path.insert(0, BASE_DIR)
from utils.config import DATA_PATH,EXPERIMENTS_DIR
from utils.inspector import inspect_row
from utils.data_loader import load_dataset
from utils.unify_columns import unify_df

In [2]:
df = pd.read_csv(os.path.join(EXPERIMENTS_DIR, "baseline_val_predictions_old.csv"))
df = unify_df(df)
df.shape

(598, 9)

In [3]:
df.head(5)

text  \
0  анализ крови по какому формеру на сифилис   
1                         стретчинг в москве   
2              стейк ресторан на белорусской   
3          дежурная стоматология севастополь   
4                                       суши   

                                             address  \
0           Санкт-Петербург, Ленинский проспект, 121   
1                    Москва, Кастанаевская улица, 39   
2           Москва, 2-й Тверской-Ямской переулок, 10   
3                 Севастополь, Античный проспект, 8В   
4  Московская область, городской округ Пушкинский...   

                                                name  \
0                                                KDL   
1                           Elasti_city; Elasti city   
2  Доктор на крыше; Doktor na kryshe; Доктор; Dok...   
3  Атмосфера; Atmosfera; Стоматология Атмосфера; ...   
4                                             Subway   

  normalized_main_rubric_name_ru     permalink  \
0        Медицинская лаборатория  200967546138   
1                    Фитнес-клуб  147723830462   
2                       Ресторан    1081999210   
3      Стоматологическая клиника   82384383021   
4                Быстрое питание  240244902340   

                                   prices_summarized  relevance  \
0  Медицинская лаборатория KDL предоставляет широ...        1.0   
1  Фитнес-клуб и студия йоги, предлагающие пробны...        0.0   
2  Ресторан «Доктор на крыше» предлагает разнообр...        0.0   
3  Стоматологическая клиника «Атмосфера» предоста...        0.0   
4  Subway предлагает широкий ассортимент сэндвиче...        0.0   

                                  reviews_summarized  pred_relevance  
0  Организация занимается проведением анализов и ...             1.0  
1  Организация занимается предоставлением услуг ф...             1.0  
2  Организация «Доктор на крыше» предлагает услуг...             0.0  
3  Организация занимается стоматологическими услу...             1.0  
4  Организация занимается быстрым питанием, предо...             0.0

In [4]:
train_data, val_data, test_data = load_dataset(DATA_PATH, drop_uncertain=True, val_frac=0.01)

In [5]:
df.relevance.value_counts()

relevance
1.0    335
0.0    263
Name: count, dtype: int64

In [6]:
df.pred_relevance.value_counts()

pred_relevance
1.0    385
0.0    213
Name: count, dtype: int64

In [7]:
acc = (df["relevance"] == df["pred_relevance"]).mean()
print("Accuracy:", round(acc, 4))

# 4. Анализ ошибок
false_positives = df[(df["relevance"] == 0.0) & (df["pred_relevance"] == 1.0)]
false_negatives = df[(df["relevance"] == 1.0) & (df["pred_relevance"] == 0.0)]

# 5. Сохранение для ручного анализа
false_positives.to_csv("false_positives.csv", index=False)
false_negatives.to_csv("false_negatives.csv", index=False)

print(f"False Positives: {len(false_positives)}")
print(f"False Negatives: {len(false_negatives)}")

Accuracy: 0.6488
False Positives: 130
False Negatives: 80


Модель склонна переоценивать релевантность — она часто предсказывает RELEVANT_PLUS, даже когда это не так. (130 FP — много ложных позитивов.)

In [6]:
# Создаем метки ошибок
df['error_type'] = 'correct'
df.loc[(df['relevance'] == 0.0) & (df['gpt_response'] == 'RELEVANT_PLUS'), 'error_type'] = 'false_positive'
df.loc[(df['relevance'] == 1.0) & (df['gpt_response'] == 'IRRELEVANT'), 'error_type'] = 'false_negative'

df[df['error_type'] == 'false_positive']
df[df['error_type'] == 'false_negative']

false_pos_df = df[df['error_type'] == 'false_positive']
false_neg_df = df[df['error_type'] == 'false_negative']

# Сохраняем отдельные ошибки
false_pos_df.to_csv("baseline_false_positives.csv", index=False)
false_neg_df.to_csv("baseline_false_negatives.csv", index=False)

In [8]:
false_pos_df['normalized_main_rubric_name_ru'].value_counts().head(10)

Кафе                                   9
Гостиница                              6
Медцентр , клиника                     6
Ресторан                               6
Автосалон                              5
Салон красоты                          3
Быстрое питание                        3
Столовая                               3
Магазин автозапчастей и автотоваров    3
Детский сад , ясли                     3
Name: normalized_main_rubric_name_ru, dtype: int64

In [9]:
false_pos_df = false_pos_df.copy()     
false_pos_df['text_length'] = false_pos_df['text'].str.len()
false_pos_df['reviews_length'] = false_pos_df['reviews_summarized'].str.len()
false_pos_df[['text_length', 'reviews_length']].describe()

text_length  reviews_length
count   130.000000       125.00000
mean     27.076923      1615.92800
std      15.217093       555.41538
min       3.000000       167.00000
25%      16.000000      1290.00000
50%      25.000000      1715.00000
75%      34.000000      1981.00000
max     100.000000      2864.00000

In [174]:
false_pos_df[false_pos_df.text_length<=27].head(100)[50:80]

text  \
477             фгос дошкольники   
478      клубные рестораны в спб   
480               печать наклеек   
481              пицца в сургуте   
484           френдс суши самара   
488           наращивание ресниц   
491                фарментин 400   
493              суши бар рязань   
504               Аутлет внуково   
519     ожоговая больница москва   
520               Нотариус 20:00   
528            3 д панорама крым   
529                   парк отель   
534      пластиковые окна калуга   
538                Доставка воды   
557      где можно рыбу половить   
571             детский психиатр   
573     лазерная эпиляция в сочи   
575   необычные рестораны москвы   
584  Мастерская по ремонту часов   
594             Жилищный адвокат   

                                               address  \
477            Москва, Трифоновская улица, 51А, стр. 2   
478   Санкт-Петербург, проспект Энгельса, 124, корп. 1   
480      Ростовская область, Батайск, улица Кирова, 9Б   
481  Ханты-Мансийский автономный округ — Югра, Сург...   
484               Самара, Молодогвардейская улица, 135   
488            Москва, проспект 60-летия Октября, 16к1   
491     Смоленская область, Сафоново, улица Ленина, 39   
493                        Рязань, Татарская улица, 95   
504  Москва, Новомосковский административный округ,...   
519          Москва, Фортунатовская улица, 1, корп. 10   
520                    Москва, Краснодонская улица, 39   
528  Республика Крым, городской округ Судак, село П...   
529        Самара, Московское шоссе, 19-й километр, 8А   
534                           Калуга, улица Ленина, 43   
538  Московская область, Одинцово, Белорусская улиц...   
557   Санкт-Петербург, Петергоф, Бобыльская дорога, 59   
571                     Москва, Валовая улица, 32/75с1   
573  Краснодарский край, Сочи, микрорайон Центральн...   
575     Москва, Космодамианская набережная, 52, стр. 6   
584  городской округ Воронеж, Ленинский район, площ...   
594          Санкт-Петербург, Владимирский проспект, 3   

                                                  name  \
477  Детский сад РЖД № 5; РЖД детский сад № 5; Детс...   
478  Хаус; Haus bar; Hausbar; Караоке бар Хаус; Рес...   
480                                 Панорама; Panorama   
481             Точка вкуса; Tochka-vkusa; Точка Вкуса   
484  Фуджи; Fuji; Friend Фуджи Суши; Friends Фуджи ...   
488       Мэйд Ю Нэйлс; Made Your Nails; MadeY; Мэйд Ю   
491                Максавит; Maksavit; Аптека Максавит   
493  Автосуши Автопицца; Avtosushi Avtopizza; Автоп...   
504                                        Högl Outlet   
519  Департамент Здравоохранения, ГКБ им. Ф. И. Ино...   
520  Нотариус Васильева Л. В.; Notarius Vasilyeva L...   
528                                 Панорама; Panorama   
529    Отель19; Hotel 19; Отель 19; Гостиница Hotel 19   
534  ТЗСК; Tulsky zavod svetoprozrachnykh konstrukt...   
538  Водомат Пейте воду; Drink The Water; Пейте вод...   
557                                  Генеральский пруд   
571  Медгород; Medgorod; Медгород клиника экспертно...   
573  Посольство красоты; Ambassade of beauty; BIOLO...   
575  City Space Bar & Restaurant; City Space; City ...   
584          Ремонт часов; Мастерская по ремонту часов   
594  Адвокат Максимов М. Н.; Advokat Maksimov M. N....   

     normalized_main_rubric_name_ru     permalink  \
477              Детский сад , ясли   29451943660   
478                    Караоке-клуб    1190255856   
480                      Фотоуслуги  242567051243   
481           Доставка еды и обедов   20536000056   
484                        Суши-бар    1327093655   
488                 Ногтевая студия   20834960525   
491                          Аптека   14020585155   
493                        Суши-бар  171672052301   
504                   Магазин обуви   61183237818   
519           Больница для взрослых   44259489914   
520                       Нотариусы    1116426080   
528                       Гостиница  17572778

In [11]:
false_pos_df[false_pos_df.text_length==100]

text  \
191  FIA07DU010 EUROBUMP Бампер задн лев с катафото...   

                          address  \
191  Москва, Кетчерская улица, 2А   

                                                  name  \
191  AvtoAll; АвтоАльянс; Автозапчасти в Новогиреев...   

          normalized_main_rubric_name_ru   permalink  \
191  Магазин автозапчастей и автотоваров  1197606102   

                                     prices_summarized  relevance  \
191  AvtoAll предлагает широкий ассортимент автозап...        0.0   

                                    reviews_summarized   gpt_response  \
191  Организация занимается продажей автозапчастей ...  RELEVANT_PLUS   

     gpt_pred_relevance      error_type  text_length  reviews_length  
191                 1.0  false_positive          100           922.0

In [190]:
train_data[train_data.text.str.contains('наращивание')]

text  \
12     наращивание ногтей екатеринбург брилиант   
140               школа наращивание ногтей свао   
2047             наращивание ресниц новокузнецк   
3619               наращивание ресниц на Выхино   
3735               наращивание ресниц на Выхино   
3962               наращивание ресниц на Выхино   
5608                         наращивание ресниц   
5758               наращивание ресниц на Выхино   
7610      обучение наращивание ресниц челябинск   
8262                         наращивание ресниц   
10855              наращивание ресниц на Выхино   
16158              наращивание ресниц на Выхино   
16332              наращивание ресниц на Выхино   
16907     обучение наращивание ресниц челябинск   
18025              наращивание ресниц на Выхино   
21518  наращивание ногтей екатеринбург брилиант   
21865                        наращивание ресниц   
22344    наращивание ресниц на пулковском шоссе   
22715                        наращивание ресниц   
23670             школа наращивание ногтей свао   
24507             школа наращивание ногтей свао   
25094    наращивание ресниц на пулковском шоссе   
25098              наращивание ресниц на Выхино   
25201            наращивание ресниц новокузнецк   
25654          мари блинкова наращивание ресниц   
26014                        наращивание ресниц   
28408          мари блинкова наращивание ресниц   
28638              наращивание ресниц на Выхино   

                                                 address  \
12     Свердловская область, Екатеринбург, проспект Л...   
140                 Москва, Берёзовая аллея, 17, корп. 2   
2047   Кемеровская область, Новокузнецк, Запорожская ...   
3619                     Москва, Святоозёрская улица, 26   
3735                     Москва, Ферганская улица, 14/13   
3962              Москва, Ферганский проезд, 10Б, стр. 2   
5608              Саратов, улица имени Т.Г. Шевченко, 54   
5758                     Москва, Вешняковская улица, 18Б   
7610               Челябинск, Комсомольский проспект, 30   
8262           Саратов, улица имени И.В. Мичурина, 55/61   
10855           Москва, Зеленодольская улица, 3, корп. 2   
16158       Московская область, Люберцы, улица Юности, 5   
16332            Москва, Большая Академическая улица, 39   
16907                    Челябинск, улица Сталеваров, 5А   
18025       Москва, Волгоградский проспект, 183, корп. 2   
21518  Свердловская область, Екатеринбург, улица Степ...   
21865                 Саратов, Первомайская улица, 47/53   
22344  Санкт-Петербург, Малая Митрофаньевская улица, ...   
22715                 Москва, Комсомольский проспект, 36   
23670                          Москва, улица Плещеева, 8   
24507                 Москва, 7-я Кожуховская улица, 15А   
25094     Санкт-Петербург, Пулковское шоссе, 71, корп. 2   
25098             Москва, Ташкентская улица, 25, корп. 1   
25201  Кемеровская область - Кузбасс, Новокузнецк, ул...   
25654                          Москва, Луговой проезд, 5   
26014               Саратов, улица имени Челюскинцев, 83   
28408  Московская область, городской округ Люберцы, р...   
28638        Москва, Лермонтовский проспект, 19, корп. 1   

                                                    name  \
12                                      Лаки; Laku; Lаки   
140                                            Top Nails   
2047                                      Agapova Studio   
3619                                             RMnails   
3735                                         Aishe Nails   
3962   Леди-Ди; Lady Dee; Салон красоты Леди-Di; Ledi...   
5608          Блэк даймонд; Black diamond; Black Diamond   
5758                           Beauty Lab; Салон красоты   
7610   Wildberries; Wildberries.ru; Wildberries.ru, п...   
8262                                         Tao Balance   
10855                             On Fleek; Блики; Bliki   
16158                  Beauty lab; Beauty Lab; Бьюти Лаб   
16332    Lashes_Val; Наращивание ресниц; Р

In [8]:
inspect_row(df, 570, pred_col="pred_relevance", label_col='relevance')

In [199]:
inspect_row(train_data, 10855)

# False Positives

In [142]:
from IPython.display import display, HTML

def inspect_row(df, idx):
    """HTML-вывод ключевых полей строки для анализа ошибок классификации."""
    row = df.loc[idx]

    html = f"""
    <div style="border:1px solid #ccc; padding:16px; border-radius:10px; font-family:sans-serif; background-color:#f9f9f9">
        <h2>📌 Index: {idx}</h2>
        <p><strong>Запрос:</strong><br>{row['text']}</p>
        <p><strong>Адрес:</strong><br>{row['address']}</p>
        <p><strong>Название:</strong><br>{row['name']}</p>
        <p><strong>Рубрика:</strong><br>{row['normalized_main_rubric_name_ru']}</p>
        <p><strong>Описание:</strong><br>{row['prices_summarized']}</p>
        <p><strong>Relevance:</strong> {row['relevance']}</p>
        <p><strong>Отзывы:</strong><br>{row['reviews_summarized']}</p>
        <p><strong>GPT ответ:</strong> {row.get('gpt_response', 'N/A')}<br>
           <strong>GPT предсказание:</strong> {row.get('gpt_pred_relevance', 'N/A')}</p>
    </div>
    """

    display(HTML(html))

## Ошибки асессоров

### ?

In [144]:
inspect_row(false_pos_df, 54)

In [145]:
inspect_row(false_pos_df,111)

In [146]:
inspect_row(false_pos_df,126)

In [147]:
inspect_row(false_pos_df,163)

In [148]:
inspect_row(false_pos_df,182)

In [149]:
inspect_row(false_pos_df,211)

In [150]:
inspect_row(false_pos_df,271)

In [151]:
inspect_row(false_pos_df,316)

In [152]:
inspect_row(false_pos_df,301)

In [153]:
train_data[((train_data['text'] == 'пиво разливное') | (train_data['text'] == 'разливное пиво')) & 
     (train_data['reviews_summarized'].str.contains('разливн.+пив', na=False)) & 
     (train_data['normalized_main_rubric_name_ru'] == 'Магазин пива')]['relevance'].value_counts()

1.0    6
0.0    2
Name: relevance, dtype: int64

Скорее, ошибка асессора судя по разбросу relevance в обучающих данных. 

In [170]:
inspect_row(false_pos_df,459)

In [178]:
inspect_row(false_pos_df, 481)

In [177]:
inspect_row(false_pos_df,484)

## Ошибки модели

#### Необходимо искать информацию об организации, в частности о часах работы
Вот тут нужно тулз, чтобы сделать запрос в гугл "адрес название организации часы работы"

In [168]:
inspect_row(false_pos_df,454)

In [154]:
inspect_row(false_pos_df,37)

In [155]:
inspect_row(false_pos_df,372)

In [171]:
inspect_row(false_pos_df,468)

In [156]:
inspect_row(false_pos_df,171)

#### Возможно, если указана какая-то конкретная организация или бренд, то нужно искать ее упоминание в предоставленной информации

In [159]:
inspect_row(false_pos_df,186)

#### Обращать внимание на нумерацию организаций

In [167]:
inspect_row(false_pos_df,414)

#### Уточнить промт - просить искать в prices_summarized, если поле не пустое. Искать конкретные услуги\товары в описании, если они есть в запросе?

In [160]:
inspect_row(false_pos_df,218)

* если запрос - какая-то медицинская процедура, то она в явном виде должна быть указана в описании организации

In [161]:
inspect_row(false_pos_df,242)

In [166]:
inspect_row(false_pos_df,403)

In [157]:
inspect_row(false_pos_df,317)

In [176]:
inspect_row(false_pos_df, 480)

### не игнорировать слово "частный"

In [172]:
inspect_row(false_pos_df, 467)

#### Если в запросе есть части адреса, тщательно проверить адрес организации

In [162]:
inspect_row(false_pos_df,355)

#### Малоинформативный запрос. 

В случае малоинформативного запроса типа "ремонт" - предсказывать по умолчанию IRRELEVANT...

In [163]:
inspect_row(false_pos_df,124)

In [164]:
inspect_row(false_pos_df,434)

In [175]:
inspect_row(false_pos_df,477)

### не ясно пока

In [143]:
inspect_row(false_pos_df, 15)

### False Negatives

## Ошибки модели

In [220]:
false_neg_df.head(20)

text  \
14   где вкусно и недорого поесть на невском проспе...   
25           где дешево поесть в санкт-петербурге 2017   
39        тетеплый плинтус 30мм высота на профиль окна   
43                                   зал для кроссфита   
64                              рестораны в тц океания   
65                        национальный парк стран мира   
69                                      автосалон лада   
71                                      текстиль метро   
79                               рестораны москва сити   
96                     франшиза итальянского ресторана   
104        стоматологическая клиника на метро аэропорт   
128                   недорогие интересные кафе москвы   
132            семейные обеды по выходным в ресторанах   
136                                 htvjyn dtkjcbgtljd   
149            пивной ресторан на соколе вместо дурдин   
150   где вкусно и недорого поесть в геленджике отзывы   
157                   итальянский ресторан профсоюзная   
158                                                чай   
169                              мрэо сетунский проезд   
170         сколько стоит капитальный ремонт двигателя   

                                               address  \
14           Санкт-Петербург, проспект Просвещения, 19   
25             Санкт-Петербург, Малая Морская улица, 7   
39   Воронежская область, Рамонский район, Яменское...   
43   Республика Башкортостан, Уфа, бульвар Тухвата ...   
64              Москва, Мироновская улица, 33, стр. 28   
65   Калужская область, Боровский район, сельское п...   
69                       Москва, Сормовская улица, 21А   
71              Москва, шоссе Энтузиастов, 12, корп. 2   
79              Москва, улица Большая Дмитровка, 7/5с2   
96                   Москва, улица Большая Полянка, 30   
104                   Москва, проезд Берёзовой Рощи, 6   
128                     Москва, улица Маросейка, 4/2с3   
132                         Москва, Цветной бульвар, 2   
136  Красноярск, проспект имени Газеты Красноярский...   
149                     Москва, Ленинский проспект, 28   
150    Краснодарский край, Новороссийск, улица Мира, 3   
157                       Москва, улица Сретенка, 26/1   
158               Москва, Люсиновская улица, 2, стр. 1   
169                     Москва, Хорошёвское шоссе, 42А   
170  Ленинградская область, Всеволожский район, Зан...   

                                                  name  \
14   Вкусно — и точка; Vkusno — i tochka; McDonald'...   
25   Буше; Bushe; БУШЕ пекарня-кондитерская ООО; Бу...   
39   Лемана Про; Lemana Pro; Leroy Merlin; Леруа Ме...   
43                                      Лейтай; Leitai   
64                 Кавказский дворик; Kavkazsky Dvorik   
65   Этнографический парк-музей Этномир; Ethnograph...   
69   АвтоГЕРМЕС сервис УАЗ; AvtoGERMES Uaz; АвтоГЕР...   
71            Домовой; Domovoj; Домовой Город; Domovoy   
79   Tehnikum; Техникум; Бистро Tehnikum; Ресторан ...   
96   Сиеста вино & паста; Siesta wine & pasta; Кули...   
104  РуДента; RuDenta; Стоматологическая клиника Ру...   
128                                Мичелада; Michelada   
132           Рыба моя; Ryba moya; Рыба моя на Цветном   
136                           Спортмастер; Sportmaster   
149                                Пивнушка; Pivnushka   
150  Вкусно — и точка; Vkusno — i tochka; McDonalds...   
157                                 Maurizio; Маурицио   
158                                           У Палыча   
169  Отделение регистрации МО ГИБДД ТНРЭР № 2 ГУ МВ...   
170                               АвтоЛидер; AvtoLider   

    normalized_main_rubric_name_ru     permalink  \
14                 Быстрое питание  209986321255   
25                         Пекарня    1023261511   
39            Строительный магазин    1583592856   
43        Спортивный клуб , секция   60891450032   
64                        Ресторан    1004953355   
65          Парк культуры и отдыха   44152522005   
69       Автосервис , автотех

## Ошибки асессоров

In [244]:
# запрос htvjyn dtkjcbgtljd
inspect_row(false_neg_df, 136)

Вопрос не к Яндекс.Картам:))

In [246]:
inspect_row(false_neg_df, 170)

Пивнушка Дурдин была по другому адресу и это вообще не метро Сокол...

In [245]:
inspect_row(false_neg_df, 149)

In [243]:
inspect_row(false_neg_df, 104)

In [242]:
inspect_row(false_neg_df, 64)

In [241]:
inspect_row(false_neg_df, 43)

In [219]:
inspect_row(false_neg_df, 14)

В отзыве четко описано "Считают цены высокими"!

In [221]:
inspect_row(false_neg_df, 25)

In [222]:
inspect_row(false_neg_df, 39)

In [249]:
train_data, val_data, test_data = load_dataset(DATA_PATH, drop_uncertain=True, val_frac=0.01)

In [252]:
test_data.isnull().sum()

text                                0
address                             0
name                                0
normalized_main_rubric_name_ru      0
permalink                           0
prices_summarized                 207
relevance                           0
reviews_summarized                 15
dtype: int64

In [29]:
train_data[train_data.text.str.contains(r'метро')]

text  \
9      новостройки москвы с отделкой от застройщика р...   
53     расположение станций метро в новой москве на к...   
103                             Ремонт глюкометров томск   
238                                          мрт в метро   
886                     суточный квартиры метро шелепиха   
...                                                  ...   
27834  сервисный центр and в москве ремонт тонометров...   
28180              Фото на документы метро академическая   
29133  Москва станция метро Парк Победы отели поблизости   
29160              недорогое кафе в москве рядом с метро   
29343           итальянский ресторан метро третьяковская   

                                                 address  \
9       Москва, Ленинградский проспект, 80к16, подъезд 3   
53     Москва, Московское центральное кольцо, станция...   
103                          Томск, проспект Фрунзе, 232   
238                      Москва, Ленинградское шоссе, 47   
886    Москва, Новомосковский административный округ,...   
...                                                  ...   
27834                       Москва, Вересковая улица, 16   
28180               Санкт-Петербург, улица Бабушкина, 12   
29133  Тульская область, Щёкинский район, сельское по...   
29160                      Москва, Яузский бульвар, 14/8   
29343              Москва, Благовещенский переулок, 10с2   

                                                    name  \
9                     Миэль; Miel; МИЭЛЬ; Офис на Соколе   
53     Крымская; Krymskaya; Кримська; Метро Крымская;...   
103                                              Дельрус   
238    Медскан; Medical Center Medskan; Medscan; Диаг...   
886    Суточная квартира; Apartments; Сутачная Кватэр...   
...                                                  ...   
27834  Ремсервисбыт; Remservisbyut; Remservisbyt; Сти...   
28180  Фотосет; FotoSet; Фото Сет; Фото копи центр; Ф...   
29133              Слобода; Sloboda; База отдыха Слобода   
29160     47+Жöн; Lozhechka; Ложечка; +47; Ложка; Lozhka   
29343            Unica; Ресторан Unica; Restaurant Unica   

              normalized_main_rubric_name_ru     permalink  \
9                     Агентство недвижимости    1227767207   
53                             Станция метро  121441159673   
103    Медицинское оборудование , медтехника   76170150859   
238                       Медцентр , клиника    1715217562   
886                          Жильё посуточно  242854723031   
...                                      ...           ...   
27834                 Ремонт бытовой техники  235281450820   
28180                             Фотоуслуги    1178912561   
29133                           Гостевой дом  153146857059   
29160                               Ресторан   20652000034   
29343                               Ресторан  198020325457   

                                       prices_summarized  relevance  \
9                                                   None        0.0   
53                                                  None        1.0   
103                                                 None        0.0   
238    Медскан предоставляет широкий спектр физиотера...        1.0   
886    Предлагает аренду однокомнатной квартиры посут...        1.0   
...                                                  ...        ...   
27834  Ремонт бытовой техники, включая стиральные, су...        0.0   
28180  Предоставляет фотоуслуги, услуги печати и копи...        0.0   
29133  Гостевой дом «Слобода» предлагает различные ва...        1.0   
29160                                               None        0.0   
29343  Ресторан Unica предлагает разнообразные блюда ...        1.0   

                                      reviews_summarized  relevance_new  
9      Организация занимается сделками с недвижимость...            0.0  
53     Организация представляет собой станцию метро М...            1.0  
103                                                 None            0

In [31]:
train_data.loc[886]

text                                               суточный квартиры метро шелепиха
address                           Москва, Новомосковский административный округ,...
name                              Суточная квартира; Apartments; Сутачная Кватэр...
normalized_main_rubric_name_ru                                      Жильё посуточно
permalink                                                              242854723031
prices_summarized                 Предлагает аренду однокомнатной квартиры посут...
relevance                                                                       1.0
reviews_summarized                Организация предоставляет жильё посуточно. Отз...
relevance_new                                                                   1.0
Name: 886, dtype: object

In [32]:
train_data.loc[886]['address']

'Москва, Новомосковский административный округ, посёлок Коммунарка, улица Потаповская Роща, 7, корп. 1'